include/alp/algorithms/symm_tridiag_eigensolver.hpp

```diff
@@ -0,0 +1,543 @@
+			print_vector( "eigensolveDiagPlusOuter: v_view ", v_view );
+#endif
+
+			// vec_b = {d_view[1], d_view[2], ... , d_view[N-1], d_view[N]+dot(v,v) }
+			size_t nn = alp::getLength( d_view );
+			alp::Vector< D > vec_b( nn );
+			auto v1 = get_view( vec_b, utils::range( 0, nn - 1 ) );
+			auto v2 = get_view( d_view, utils::range( 1, nn ) );
+			rc = rc ? rc : alp::set( v1, v2 );
+			auto v3 = get_view( vec_b, utils::range( nn - 1, nn ) );
+			auto v4 = get_view( d_view, utils::range( nn - 1, nn ) );
+			rc = rc ? rc : alp::set( v3, v4 );
+
+			// eWiseLambda currently does not work with select view
+			// dot() does not work with select view
+			// as a (temp) solution we use temp vectors
+			alp::Vector< D > vec_temp_egvals( nn );
+			alp::Vector< D > vec_temp_d( nn );
+			alp::Vector< D > vec_temp_v( nn );
+
+			rc = rc ? rc : alp::set( vec_temp_egvals, zero );
+			rc = rc ? rc : alp::set( vec_temp_d, d_view );
+			rc = rc ? rc : alp::set( vec_temp_v, v_view );
+
+			Scalar< D > alpha( zero );
+			// there is a bug in dot() when called on select views
+			//rc = rc ? rc : dot( alpha, d_view, d_view, ring );
+			rc = rc ? rc : dot( alpha, vec_temp_v, vec_temp_v, ring );
+
+			auto v5 = get_view( vec_b, utils::range( alp::getLength( vec_b ) - 1, alp::getLength( vec_b ) ) );
+			rc = rc ? rc : alp::foldl( v5, alpha, ring.getAdditiveOperator() );
+
+			rc = rc ? rc : eWiseLambda(
+				[ &d_view, &vec_temp_v, &vec_b ]( const size_t i, D &val ) {
+					Scalar< D > a( d_view[ i ] );
+					Scalar< D > b( vec_b[ i ] );
+					Scalar< D > w( ( *a + *b ) / 2 );
+					bisec_sec_eq( w, d_view, vec_temp_v, a, b );
+					val = *w;
+				},
+				vec_temp_egvals
+			);
+			rc = rc ? rc : alp::set( egvals_non_direct, vec_temp_egvals );
+
+			Matrix< D, structures::General, Dense > tmp_egvecs( nn, nn );
+			Matrix< D, structures::General, Dense > tmp_denominator( nn, nn );
+
+			alp::Vector< D > ones( nn );
+			rc = rc ? rc : alp::set( ones, one );
+			rc = rc ? rc : alp::set(
+				tmp_egvecs,
+				alp::outer( vec_temp_v, ones, ring.getMultiplicativeOperator() )
+			);
+
+			auto ddd = alp::outer( vec_temp_d, ones, ring.getMultiplicativeOperator() );
+			auto lll = alp::outer( ones, egvals_non_direct, ring.getMultiplicativeOperator() );
+			rc = rc ? rc : alp::set( tmp_denominator, ddd );
+			rc = rc ? rc : alp::foldl( tmp_denominator, lll, minus );
+			rc = rc ? rc : alp::foldl( tmp_egvecs, tmp_denominator, divide );
+
+			// while fold matrix -> vector would be a solution to
+			// normalize columns in tmp_egvecs,
+			// here we abuse the syntax and use eWiseLambda.
+			// Once fold matrix -> vector implemented, the next section should be rewritten
+			rc = rc ? rc : eWiseLambda(
+				[ &tmp_egvecs, &nn, &ring, &divide, &zero ]( const size_t i, D &val ) {
+					(void)val;
+					auto egvec_i = get_view( tmp_egvecs, utils::range( 0, nn ), i );
+					Scalar< D > norm_i( zero );
+					(void)norm2( norm_i, egvec_i, ring );
```

Comment:
Why is there `(void)` here but not for most of other calls to ALP operations? Additionally, should we check the correctness of each call to ALP primitives?

Reply:
I think we do not check `rc` within `eWiseLamba` ? The void on `norm2` is not needed.